ludwig/models/llm.py

```diff
@@ -222,12 +222,37 @@ def initialize_adapter(self):
                     "`finetune` or remove the adapter config."
                 )
 
-            from peft import get_peft_model, TaskType
+            from peft import get_peft_model
+
+            if self.config_obj.adapter.pretrained_adapter_weights:
+                logger.info(f"Using pretrained adapter weights: {self.config_obj.adapter.pretrained_adapter_weights}")
+                # If pretrained adapter weights are provided, we want to load them into the model
+                from peft import MODEL_TYPE_TO_PEFT_MODEL_MAPPING, PeftConfig
+
+                peft_config = PeftConfig.from_pretrained(self.config_obj.adapter.pretrained_adapter_weights)
+                peft_dict = peft_config.to_dict()
+
+                # Need to update the peft config with some of the values from config_obj because not all of them are set
+                for param_name, param_value in self.config_obj.adapter.to_config().to_dict().items():
+                    # Not all parameters are supported by all models, so we only add the parameter to the load kwargs
+                    # if it is supported by the model.
+                    if param_value is None:
```

Comment:
Could you add a comment with examples for when this happens?

When are `param_value` from `self.config_obj.adapter.to_config().to_dict().items()` `None` and why should we skip adding them to the peft_dict?

When is the `param_name` not in peft_dict that comes from HF? (Is it just the `type` parameter?

Reply:
The newly added comment LGTM. @Infernaught could you mark this comment as resolved?